examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    return null;
+  }
+
+  public final String getArgumentName(int index) {
+    if (index < mArgs.size()) {
+      return mArgs.get(index).name;
+    }
+
+    return null;
+  }
+
+  public final int getArgumentsCount() {
+    return mArgs.size();
+  }
+
+  public final String getArgumentDescription(int index) {
+    if (index < mArgs.size()) {
+      return mArgs.get(index).desc;
+    }
+
+    return null;
+  }
+
+  public final int addArgument(String name, StringBuffer out) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Attribute;
+    arg.name = name;
+    arg.value = (Object) out;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final int addArgument(String name, int min, int max, MutableInteger out) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Number_int32;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.min = min;
+    arg.max = max;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final int addArgument(String name, long min, long max, MutableInteger out) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Number_int64;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.min = min;
+    arg.max = max;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final int addArgument(String name, IPAddress out) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Address;
+    arg.name = name;
+    arg.value = (Object) out;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final int addArgument(String name, String out, String desc) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.String;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.desc = desc;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final int addArgument(String name, long min, long max, Optional<Long> out, String desc) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Number_int64;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.min = min;
+    arg.max = max;
+    arg.flags = Argument.kOptional;
+    arg.desc = desc;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final int addArgument(
+      String name, short min, short max, Optional<Short> out, String desc) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Number_int16;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.min = min;
+    arg.max = max;
+    arg.flags = Argument.kOptional;
+    arg.desc = desc;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final int addArgument(String name, Optional<Boolean> out, String desc) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Bool;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.flags = Argument.kOptional;
+    arg.desc = desc;
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final boolean initArguments(int argc, String[] args) {
```

Comment:
need explanation of what `initArguments` returning a bool means.

It is unclear to me why this sometimes returns false but on other times it does a System.exit()